objects_01.ts

```diff
@@ -0,0 +1,69 @@
+class Warrior {
+    public readonly name: string;
+    public readonly damage: number;
+    public hitPoints: number;
+ 
+    constructor(name: string, damage: number, hitPoints: number) {
+        this.name = name;
+        this.damage = damage;
+        this.hitPoints = hitPoints;
+    }
+    attack(defender: Warrior):void {
+        defender.hitPoints -= this.damage;
+    }
+}
+
+class Battle {
+    private teams: Warrior[][];
+    private messages: Array<string>;
```

Comment:
Одинаково лучше делать всегда, либо через Array<>, либо так

```suggestion
    private messages: string[];
```